In [1]:
import pandas as pd
import numpy as np
import geopandas
from shapely.geometry import Point

In [2]:
burnt = pd.read_pickle('../../finalproj_data/csv/burnt.pkl')

In [3]:
burnt

lat         lon  Burn_Date
0           4.360417   96.327083        0.0
1           4.360417   96.331250        0.0
2           4.360417   96.335417        0.0
3           4.360417   96.339583        0.0
4           4.360417   96.343750        0.0
...              ...         ...        ...
210892456 -48.231250  165.918750       -2.0
210892457 -48.231250  165.922917       -2.0
210892458 -48.231250  165.927083       -2.0
210892459 -48.231250  165.931250       -2.0
210892460 -48.231250  165.935417       -2.0

[210892461 rows x 3 columns]

Unique values:
    array([  0.,  nan,  -2., 305., 307., 322., 327., 320., 323., 326., 328.,
       324., 321., 319., 329., 313., 316., 306., 308., 309., 317., 312.,
       311., 333., 334., 332., 331., 310., 325., 318., 314., 315., 330.] <br>
14.6% of 210,892,461 data points are null values. <br>
0 is unburned, -2 is water, other values are days

In [11]:
burnt.columns

Index(['lat', 'lon', 'Burn_Date'], dtype='object')

In [ ]:
# Change to geoDataFrame
gdf = geopandas.GeoDataFrame(burnt, geometry=geopandas.points_from_xy(burnt.lon, burnt.lat))

In [ ]:
# Check on a plot to see what it looks like

world = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))

# We restrict to Australia
ax = world[world.continent == 'Australia'].plot(
    color='white', edgecolor='black')

# We can now plot our ``GeoDataFrame``.
gdf.plot(ax=ax, color='red')

plt.show()